CMakeLists.txt

```diff
@@ -1160,12 +1160,8 @@ if(CONFIG_USERSPACE AND CONFIG_ARM)
   add_dependencies(     priv_stacks_prebuilt linker_priv_stacks_script ${OFFSETS_LIB})
 endif()
 
-# FIXME: Is there any way to get rid of empty_file.c?
-add_executable(       ${ZEPHYR_PREBUILT_EXECUTABLE} misc/empty_file.c)
-target_link_libraries(${ZEPHYR_PREBUILT_EXECUTABLE} ${TOPT} ${PROJECT_BINARY_DIR}/linker.cmd ${PRIV_STACK_LIB} ${zephyr_lnk} ${CODE_RELOCATION_DEP})
-set_property(TARGET   ${ZEPHYR_PREBUILT_EXECUTABLE} PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker.cmd)
-add_dependencies(     ${ZEPHYR_PREBUILT_EXECUTABLE} ${PRIV_STACK_DEP} ${LINKER_SCRIPT_TARGET} ${OFFSETS_LIB})
-
```

Comment:
I don't see anything toolchain-specific in this code.

Reply:
TOPT, linker.cmd, and the order in which the arguments are given to target_link_libraries.

Reply:
We can make TOPT evaluate to something that any toolchain will accept, e.g. the empty string.

I agree about linker.cmd, I would suggest introducing a variable for ${PROJECT_BINARY_DIR}/linker.cmd, and have that variable point to something appropriate for the toolchain.

How is the order of arguments toolchain-specific?